api_tests/lib/actor.ts

```diff
@@ -69,35 +78,113 @@ export class Actor {
         return response.body.id;
     }
 
-    pollComitNodeUntil(
+    async pollComitNodeUntil(
         location: string,
         predicate: (body: Entity) => boolean
     ): Promise<Entity> {
-        return new Promise((final_res, rej) => {
-            request(this.comit_node_url())
-                .get(location)
-                .end((err, res) => {
-                    if (err) {
-                        return rej(err);
-                    }
-                    res.should.have.status(200);
-
-                    if (predicate(res.body)) {
-                        final_res(res.body);
-                    } else {
-                        setTimeout(async () => {
-                            const result = await this.pollComitNodeUntil(
-                                location,
-                                predicate
-                            );
-                            final_res(result);
-                        }, 500);
-                    }
-                });
-        });
+        let response = await request(this.comit_node_url()).get(location);
+
+        expect(response).to.have.status(200);
+
+        if (predicate(response.body)) {
+            return response.body;
+        } else {
+            await sleep(500);
+
+            return this.pollComitNodeUntil(location, predicate);
+        }
+    }
+
+    async doComitAction(action: Action) {
+        let { url, body, method } = this.buildRequestFromAction(action);
+
+        let agent = request(this.comit_node_url());
+        let response;
+
+        // let's ditch this stupid HTTP library ASAP to avoid this ...
```

Comment:
Is the only reason why you complain the `switch` below or because of something else? 

In general I agree as I have the feeling we _outgrew_ the purpose of this library. It's a plugin for chai and meant to be used for testing while we basically use it as a http client.

Reply:
Yes mainly because of the `switch` because there was no way of passing the HTTP method name to the request ...